beluga_example/launch/utils/ndt_localization_launch.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2024 Ekumen, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+
+from ament_index_python.packages import get_package_share_directory
+
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.actions import GroupAction
+from launch_ros.actions import ComposableNodeContainer
+from launch_ros.actions import Node
+from launch_ros.descriptions.composable_node import ComposableNode
+from launch.utilities.type_utils import get_typed_value
+
+from launch_ros.actions import SetParameter
+
+from beluga_example.launch_utils import with_launch_arguments
+from beluga_example.launch_utils import (
+    get_node_with_arguments_declared_from_params_file,
+)
+
+
+def get_launch_arguments():
+    example_dir_path = Path(get_package_share_directory('beluga_example'))
+    return [
+        DeclareLaunchArgument(
+            name='localization_plugin',
+            default_value='beluga_amcl::NdtAmclNode',
+            description='Localization node plugin to use if composition is enabled. ',
+        ),
+        DeclareLaunchArgument(
+            name='localization_prefix',
+            default_value='',
+            description='Set of commands/arguments to precede the node command (e.g. "timem --").',
+        ),
+        DeclareLaunchArgument(
+            name='localization_params_file',
+            default_value=str(example_dir_path / 'params' / 'default_ndt.ros2.yaml'),
+            description='Parameters file to be used to run the localization node.',
+        ),
+        DeclareLaunchArgument(
+            name='localization_map',
+            default_value=str(example_dir_path / 'maps' / 'turtlebot3_world.yaml'),
+            description='Map YAML file to be used by the map server node.',
+        ),
+        DeclareLaunchArgument(
+            name='ndt_map',
+            default_value=str(example_dir_path / 'maps' / 'turtlebot3_world.hdf5'),
+            description='Map HDF5 file used by the localization node.',
+        ),
```

Comment:
@serraramiro1 meta: do we need to pass the `ndt_map` explicitly? `get_node_with_arguments_declared_from_params_file` already declares arguments based on ROS 2 parameters YAML file contents.

Reply:
Unfortunately this way of passing parameters doesn't support variable substituion, that's why I went for this approach.
Happy to use another

Reply:
That's a good point. That magic is incomplete. Oh well, nevermind.